src/program/snabbvmx/query/query.lua

```diff
@@ -109,19 +109,23 @@ function run (raw_args)
    parse_args(raw_args)
    print("<snabb>")
    local pids = {}
+   local pids_name = {}
    for _, pid in ipairs(shm.children("/")) do
      if shm.exists("/"..pid.."/nic/id") then
        local lwaftr_id = shm.open("/"..pid.."/nic/id", lwtypes.lwaftr_id_type)
-       local instance_id = ffi.string(lwaftr_id.value)
+       local instance_id_name = ffi.string(lwaftr_id.value)
+       local _, _, instance_id = string.find(instance_id_name, "(%d+)")
```

Comment:
As a suggestion I think it's more readable to use match to get the id value, like:

```lua
local instance_id = instance_id_name:match("(%d+)")
```